tests/engine/processes/calcjobs/test_calc_job.py

```diff
@@ -874,11 +874,12 @@ class TestImport:
     """Test the functionality to import existing calculations completed outside of AiiDA."""
 
     @pytest.fixture(autouse=True)
-    def init_profile(self, aiida_profile_clean, aiida_localhost):  # pylint: disable=unused-argument
+    def init_profile(self, aiida_profile_clean, aiida_localhost, aiida_local_code_factory):  # pylint: disable=unused-argument
         """Initialize the profile."""
         # pylint: disable=attribute-defined-outside-init
         self.computer = aiida_localhost
         self.inputs = {
+            'code': aiida_local_code_factory('core.arithmetic.add', '/bin/bash', computer=aiida_localhost),
```

Comment:
Just out of curiosity. I think the `code` is mandatory input for calcjob. This is not in the original code, but why did that not fail the test?

Reply:
It is in fact not required in the port. For normal executions it is sort of required because the validator will make sure it is defined. However, there is an exception for imported jobs. See these lines https://github.com/aiidateam/aiida-core/blob/9d903ca2c73ef0e7e967c17ff3d7eac3410f6b58/aiida/engine/processes/calcjobs/calcjob.py#L57-L63

If a `remote_folder` is defined, it is an import job and so the code is not required.